pkg/api/meta/multirestmapper.go

```diff
@@ -169,24 +170,29 @@ func (m MultiRESTMapper) RESTMapping(gk unversioned.GroupKind, versions ...strin
 	if len(allMappings) == 1 {
 		return allMappings[0], nil
 	}
+	if len(allMappings) > 1 {
+		return nil, fmt.Errorf("multiple matches found for %v in %v", gk, versions)
+	}
 	if len(errors) > 0 {
 		return nil, utilerrors.NewAggregate(errors)
 	}
-	if len(allMappings) == 0 {
-		return nil, fmt.Errorf("no match found for %v in %v", gk, versions)
-	}
-
-	return nil, fmt.Errorf("multiple matches found for %v in %v", gk, versions)
+	return nil, fmt.Errorf("no match found for %v in %v", gk, versions)
 }
 
 // AliasesForResource finds the first alias response for the provided mappers.
 func (m MultiRESTMapper) AliasesForResource(alias string) ([]string, bool) {
+	seenAliases := sets.NewString()
 	allAliases := []string{}
 	handled := false
 
 	for _, t := range m {
 		if currAliases, currOk := t.AliasesForResource(alias); currOk {
-			allAliases = append(allAliases, currAliases...)
+			for _, currAlias := range currAliases {
```

Comment:
Related to https://github.com/kubernetes/kubernetes/issues/22337.

Shall we allow different RESTMappers to have same aliases? I think we should just make the resources that got aliased to be qualified with group (https://github.com/kubernetes/kubernetes/blob/master/pkg/api/install/install.go#L90), then we don't need the logic here.

And currently you see duplication because the aliases are stored in a global variable (https://github.com/kubernetes/kubernetes/blob/master/pkg/api/meta/restmapper.go#L492), we should make it a member variable of DefaultRESTMapper.